src/Microsoft.Unity.Analyzers/SetPositionAndRotation.cs

```diff
@@ -0,0 +1,241 @@
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *-------------------------------------------------------------------------------------------*/
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.Unity.Analyzers.Resources;
+
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.Unity.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class SetPositionAndRotationAnalyzer : DiagnosticAnalyzer
+	{
+		internal static readonly DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+			id: "UNT0022",
+			title: Strings.SetPositionAndRotationDiagnosticTitle,
+			messageFormat: Strings.SetPositionAndRotationDiagnosticMessageFormat,
+			category: DiagnosticCategory.Performance,
+			defaultSeverity: DiagnosticSeverity.Info,
+			isEnabledByDefault: true,
+			description: Strings.SetPositionAndRotationDiagnosticDescription);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.RegisterSyntaxNodeAction(AnalyzeAssignment, SyntaxKind.SimpleAssignmentExpression);
+		}
+
+		private static void AnalyzeAssignment(SyntaxNodeAnalysisContext context)
+		{
+			if (!(context.Node is AssignmentExpressionSyntax))
+				return;
+
+			var assignmentExpression = (AssignmentExpressionSyntax)context.Node;
+
+			if (!IsSetPositionOrRotation(assignmentExpression, context.SemanticModel))
+				return;
+
+
+			if (context.Node.FirstAncestorOrSelf<BlockSyntax>() == null)
+				return;
+
+			var block = context.Node.FirstAncestorOrSelf<BlockSyntax>();
+
+			if (context.Node.FirstAncestorOrSelf<ExpressionStatementSyntax>() == null)
+				return;
+
+			var expression = context.Node.FirstAncestorOrSelf<ExpressionStatementSyntax>();
+
+			var siblingsAndSelf = block.ChildNodes().ToImmutableArray();
+
+			if (siblingsAndSelf.LastIndexOf(expression) == -1)
+				return;
+
+			var currentIndex = siblingsAndSelf.LastIndexOf(expression);
+
+			var nextIndex = currentIndex + 1;
+
+			if (nextIndex == siblingsAndSelf.Length)
+				return;
+
+			var statement = siblingsAndSelf[nextIndex];
+
+			if (!(statement is ExpressionStatementSyntax))
+				return;
+
+			var nextExpression = ((ExpressionStatementSyntax)statement).Expression;
+
+
+			if (!(nextExpression is AssignmentExpressionSyntax))
+				return;
+
+			var nextAssignmentExpression = (AssignmentExpressionSyntax)nextExpression;
```

Comment:
Pattern matching:

```suggestion
			if (!(statement is ExpressionStatementSyntax expressionStatement))
				return;

			if (!(expressionStatement.Expression is AssignmentExpressionSyntax nextAssignmentExpression))
				return;
```

See how it is more concise? Everytime is/as/cast operators are used, think if you cannot use pattern matching construct instead.